include/votca/csg/topologymap.h

```diff
@@ -30,26 +31,26 @@ namespace csg {
 
 class TopologyMap {
  public:
-  ~TopologyMap();
-
   TopologyMap(Topology *in, Topology *out);
 
-  void AddMoleculeMap(Map *map);
+  void AddMoleculeMap(std::unique_ptr<Map> map);
 
   void Apply();
 
  private:
   Topology *_in;
   Topology *_out;
 
-  using MapContainer = std::vector<Map *>;
+  using MapContainer = std::vector<std::unique_ptr<Map>>;
```

Comment:
why not just make it

```suggestion
  using MapContainer = std::vector<Map>;
```

Reply:
and make it value based.